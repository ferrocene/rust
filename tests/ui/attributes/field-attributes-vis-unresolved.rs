//@ revisions: edition2015 edition2018
//@[edition2015]edition:2015
//@[edition2018]edition:2018..
// Non-builtin attributes do not mess with field visibility resolution (issue #67006).

mod internal {
    struct S {
        #[rustfmt::skip]
        pub(in crate::internal) field: u8 // OK
    }

    struct Z(
        #[rustfmt::skip]
        pub(in crate::internal) u8 // OK
    );
}

struct S {
    #[rustfmt::skip]
    pub(in nonexistent) field: u8 //[edition2015]~ ERROR failed to resolve
    //[edition2018]~^ ERROR relative paths are not supported in visibilities in 2018 edition or later
}

struct Z(
    #[rustfmt::skip]
    pub(in nonexistent) u8 //[edition2015]~ ERROR failed to resolve
    //[edition2018]~^ ERROR relative paths are not supported in visibilities in 2018 edition or later
);

fn main() {}
