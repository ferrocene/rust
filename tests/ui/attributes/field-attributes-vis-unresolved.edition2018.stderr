error: relative paths are not supported in visibilities in 2018 edition or later
  --> $DIR/field-attributes-vis-unresolved.rs:20:12
   |
LL |     pub(in nonexistent) field: u8
   |            ^^^^^^^^^^^ help: try: `crate::nonexistent`

error: relative paths are not supported in visibilities in 2018 edition or later
  --> $DIR/field-attributes-vis-unresolved.rs:26:12
   |
LL |     pub(in nonexistent) u8
   |            ^^^^^^^^^^^ help: try: `crate::nonexistent`

error: aborting due to 2 previous errors

